src/APL/fileAPL.ts

```diff
@@ -0,0 +1,87 @@
+import debugPkg from "debug";
+import { promises as fsPromises } from "fs";
+
+import { APL, AuthData } from "./apl";
+
+const debug = debugPkg.debug("FileAPL");
+
+export const loadDataFromFile = async (fileName: string): Promise<AuthData | undefined> => {
+  debug(`Load auth data from the ${fileName} file`);
+  let parsedData: Record<string, string> = {};
+  try {
+    await fsPromises.access(fileName);
+    parsedData = JSON.parse(await fsPromises.readFile(fileName, "utf-8"));
+  } catch (err) {
+    debug(`Could not read auth data from the ${fileName} file`, err);
+    return undefined;
+  }
+  const { token, domain } = parsedData;
+  if (token && domain) {
+    return { token, domain };
+  }
+  return undefined;
+};
+
+export const saveDataToFile = async (fileName: string, authData?: AuthData) => {
+  debug(`Save auth data to the ${fileName} file`);
+  const newData = authData ? JSON.stringify(authData) : "{}";
+  try {
+    await fsPromises.writeFile(fileName, newData);
+  } catch (err) {
+    debug(`Could not save auth data to the ${fileName} file`, err);
+  }
```

Comment:
Error is hidden silently here. App can't work if the file is not writeable. So as a user I'd expect it to fail fast - throw as soon as possible, telling me that I must ensure permissions are correct. Otherwise there will be hidden bugs why auth doesn't work